core/src/server/ValidationUtil.cpp

```diff
@@ -116,7 +116,7 @@ ValidateCollectionName(const std::string& collection_name) {
     int64_t table_name_size = collection_name.size();
     for (int64_t i = 1; i < table_name_size; ++i) {
         char name_char = collection_name[i];
-        if (name_char != '_' && std::isalnum(name_char) == 0) {
+        if (name_char != '_' && name_char != '$' && std::isalnum(name_char) == 0) {
```

Comment:
suggest to define a global unordered_set:
  const unordered_set<char> valid_collection_char_set = {'_', '$'};

Reply:
I don't think it is necessary to define 'valid_collection_char_set' since [0-9,a-z,A-Z] are all valid.
Furthermore, what is a valid name may be changed again and again……